src/test/scala/firrtlTests/annotationTests/JsonProtocolSpec.scala

```diff
@@ -18,6 +20,15 @@ case class AnAnnotation(
     groundType: GroundType
 ) extends NoTargetAnnotation
 
+class AnnoInjector extends Transform with DependencyAPIMigration with PreservesAll[Transform] {
+  override def optionalPrerequisiteOf = Dependency[ChirrtlEmitter] :: Nil
+  def execute(state: CircuitState): CircuitState = {
+    // Classes defined in method bodies can't be serialized by json4s
+    case class MyAnno(x: Int) extends NoTargetAnnotation
+    state.copy(annotations = MyAnno(3) +: state.annotations)
+  }
+}
+
```

Comment:
It's better if the PR overrides `invalidates` instead of mixes in `PreservesAll` (as this may go away).